prime-router/src/main/kotlin/fhirengine/engine/FHIRDestinationFilter.kt

```diff
@@ -0,0 +1,561 @@
+        this.actionLogger = actionLogger
+
+        message as ReportPipelineMessage
+
+        // track input report
+        actionHistory.trackExistingInputReport(message.reportId)
+
+        // pull fhir document and parse FHIR document
+        val fhirJson = message.downloadContent()
+        val bundle = FhirTranscoder.decode(fhirJson)
+
+        // get the receivers that this bundle should go to
+        val listOfReceivers = findReceiversForBundle(bundle, message.reportId, actionHistory, message.topic)
+
+        // go up the report lineage to get the sender of the root report
+        val sender = reportService.getSenderName(message.reportId)
+
+        // send event to Azure AppInsights
+        val observationSummary = AzureEventUtils.getObservations(bundle)
+        azureEventService.trackEvent(
+            ReportAcceptedEvent(
+                message.reportId,
+                message.topic,
+                sender,
+                observationSummary,
+                fhirJson.length
+            )
+        )
+
+        // check if there are any receivers
+        if (listOfReceivers.isNotEmpty()) {
+            return listOfReceivers.flatMap { receiver ->
+                val sources = emptyList<Source>()
+                val report = Report(
+                    Report.Format.FHIR,
+                    sources,
+                    1,
+                    metadata = this.metadata,
+                    topic = message.topic,
+                    destination = receiver
+                )
+
+                // create item lineage
+                report.itemLineages = listOf(
+                    ItemLineage(
+                        null,
+                        message.reportId,
+                        1,
+                        report.id,
+                        1,
+                        null,
+                        null,
+                        null,
+                        report.getItemHashForRow(1)
+                    )
+                )
+
+                val nextEvent = ProcessEvent(
+                    Event.EventAction.RECEIVER_FILTER,
+                    report.id,
+                    Options.None,
+                    emptyMap(),
+                    emptyList()
+                )
+
+                // upload new copy to blobstore
+                val bodyString = FhirTranscoder.encode(bundle)
+                val blobInfo = BlobAccess.uploadBody(
+                    Report.Format.FHIR,
+                    bodyString.toByteArray(),
+                    report.name,
+                    message.blobSubFolderName,
+                    nextEvent.eventAction
+                )
+                // ensure tracking is set
+                actionHistory.trackCreatedReport(nextEvent, report, blobInfo = blobInfo)
+
+                val receiverObservationSummary = AzureEventUtils.getObservations(bundle)
+                azureEventService.trackEvent(
+                    ReportRouteEvent(
```

Comment:
I don't think we want to dispatch this here?  As currently written this is what should get dispatched at the end of the FHIRReceiverFilter unless I'm missing something here